v0/pkg/prometheus/gen/collector_test.go

```diff
@@ -117,8 +119,13 @@ func TestUpdateSimpleTransformation(t *testing.T) {
 }
 
 func TestUpdateFull(t *testing.T) {
-	col, upd := NewRecursiveMetricsFromTags(&full)
+	col, upd := NewRecursiveMetricsFromTags(&full, WithMetricNameTransform(
```

Comment:
I think it would be better to add a new Test, so we test the metric name transform separately.

Reply:
This wont work as we need some basic metric name transformation. I removed the metricNameCleaner in internal/collector/collector.go so the default test cant work.

Reply:
I had to add it to even more tests to get them working